### PR TITLE
Move compute kernel config to TTNN

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue : #11645 

Current Status: PR Review
### Checklist
- [x] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600245128)
- [x] [(Single-card) Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/runs/10600277230)
- [x] [[post-commit] models tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600330619) 
- [x] [(Single-card) Model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600247196)
- [x] [(T3K) T3000 model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/10594341198) 
- [x] [(T3K) T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600259623)
- [x] [(T3K) T3000 profiler tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600257788)
- [x] [(T3K) T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600253056)
- [x] [Nightly fast dispatch tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600265426)
- [x] [(T3K) T3000 frequent tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600249193)
- [x] [(Single-card) Demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/10600282305)

### CODE REVIEW

File: ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/layernorm_distributed/layernorm_post_allgather_op.hpp
Comment (yan-zaretskiy) [inline ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/layernorm_distributed/layernorm_post_allgather_op.hpp:1]: I think this file is meant to be deleted, as well as the next one.

Comment (Aswinmcw) [reply ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/layernorm_distributed/layernorm_post_allgather_op.hpp:1]: > I think this file is meant to be deleted, as well as the next one.

Yes I will remove it

File: ttnn/cpp/pybind11/operations/core.hpp [hunk 1]
```diff
@@ -282,6 +282,30 @@ void py_module(py::module& module) {
             py::arg("dtype") = std::nullopt,
             py::arg("memory_config") = std::nullopt,
             py::arg("device") = nullptr});
+
+    py::class_<DeviceComputeKernelConfig>(module, "DeviceComputeKernelConfig");
```
Comment (KalaivaniMCW) [inline ttnn/cpp/pybind11/operations/core.hpp:286]: have the class binding under a separate function py_module_types, above py_module

File: ttnn/cpp/pybind11/operations/__init__.hpp [hunk 2]
```diff
@@ -48,6 +47,9 @@ namespace ttnn {
 namespace operations {
 
 void py_module(py::module& module) {
+    auto m_core = module.def_submodule("core", "core operations");
```
Comment (KalaivaniMCW) [inline ttnn/cpp/pybind11/operations/__init__.hpp:50]: add core::py_module_types

File: ttnn/cpp/pybind11/operations/core.hpp (hunk 1 above)
Comment (Aswinmcw) [reply ttnn/cpp/pybind11/operations/core.hpp]: Done

File: ttnn/cpp/pybind11/operations/__init__.hpp (hunk 2 above)
Comment (Aswinmcw) [reply ttnn/cpp/pybind11/operations/__init__.hpp:50]: Done

**Review by KalaivaniMCW — APPROVED** [review]
Approved

**Review by yan-zaretskiy — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]

**Review by yugaoTT — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Review by mtairum — APPROVED** [review]
All good assuming that the tests pass.

**Review by bbradelTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by esmalTT — APPROVED** [review]

**Review by johanna-rock-tt — APPROVED** [review]

**Review by ntarafdar — APPROVED** [review]

**Pull request merged**